src/systems/ClockSystem.cpp

```diff
@@ -0,0 +1,18 @@
+#include "../pch.h"
+#include "ClockSystem.h"
+#include "../components/world/ClockComponent.h"
+
+ClockSystem::ClockSystem(entt::registry &registry)
+    : m_registry(registry)
+{}
+
+void ClockSystem::update(float deltaTime)
+{
+    auto clockView = m_registry.view<ClockComponent>();
+    if (clockView.empty()) return;
+
+    auto clock = m_registry.view<ClockComponent>()[0];
```

Comment:
i've seen we search with entity's name and not selecting the first entity with the required component in most places. maybe we should change it to something like this everywhere or use the old fashioned way here

Reply:
I did this because we can have only one clock in the game